wgpu-core/src/device/mod.rs

```diff
@@ -204,7 +212,8 @@ impl<B: GfxBackend> Device<B> {
         mem_props: hal::adapter::MemoryProperties,
         non_coherent_atom_size: u64,
         supports_texture_d24_s8: bool,
-        max_bind_groups: u32,
+        desc: &wgt::DeviceDescriptor,
+        #[cfg(feature = "trace")] trace_path: Option<&std::path::Path>,
```

Comment:
Same here, I think it would be better to always have the argument and just pass `None` when `trace` isn't passed